contrib/building.md

```diff
@@ -0,0 +1,280 @@
+# Working With Realm-JS
+
+
+<!--ts-->
+  * [Install Instructions For Mac](#instructions-for-macos)
+  * [Hello World Projects To Practice](#test-projects)
+  * [Building Realm for React Native **Android**](#building-for-android)
+  * [Building Realm for React Native **iOS**](#ios)
+  * [Building Realm for **NodeJS**](#building-for-nodejs)
+
+<!--te-->
+
+
+## Pre-Requisites
+
+Clone [realm-js project](https://github.com/realm/realm-js) and install the required dependencies:
+
+* XCode 9.4+
+* [NodeJS](https://nodejs.org/en/) version 10 or later
+* Android SDK 23+
+* Android NDK 21.0
+  - Available via the SDK Manager on Android Studio **Tools > SDK Manager**.
+
+### Instructions for MacOS:
+
+```sh
+  #Install brew
+  bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+
+  #Install NVM, OpenJDK 8 and cocoapods
+  brew install nvm openjdk@8 cocoapods
+
+```
+
+Android Setup to build from the command-line:
+
+```sh
+# Before following this make sure java is available in the command line.
+# Location of your Android SDK
+export ANDROID_HOME=$HOME/Library/Android/sdk
```

Comment:
The `ANDROID_HOME` environment variable [is deprecated](https://developer.android.com/studio/command-line/variables#android_sdk_root).
This is what I added to my ~/.zshrc

```suggestion
# Export Android SDK tools
# See https://stackoverflow.com/questions/23042638/how-do-i-set-android-sdk-home-environment-variable
export ANDROID_SDK_ROOT=$HOME/Library/Android/sdk
export ANDROID_SDK_HOME=$HOME/.android
export ANDROID_EMULATOR_HOME=$HOME/.android
export ANDROID_AVD_HOME=$HOME/.android/avd
# export ANDROID_NDK_ROOT=
export PATH=$PATH:$ANDROID_SDK_ROOT/tools
export PATH=$PATH:$ANDROID_SDK_ROOT/platform-tools
```

It would be great to add a note about adding this to the ~/.zshrc too.